src/compas/geometry/transformations/quaternions.py

```diff
@@ -0,0 +1,317 @@
+"""
+This module contains functions that operate on and/or return quaternions.
+
+Default convention to represent a quaternion in this module is a list of four real values [qw, qx, qy, qz].
+The first value qw is the scalar (real) part, and qx,qy,qz form the vector (complex, imaginary) part, so that:
+q = qw + qx*i + qy*j + qz*k,
+where i,j,k are basis components.
+
+Quaternion algebra basics:
+i*i = j*j = k*k = i*j*k = -1
+i*j = k    j*i = -k
+j*k = i    k*j = -i
+k*i = j    i*k = -j
+
+Quaternion as rotation:
+A rotation through an angle theta around an axis defined by a euclidean unit vector u = ux*i + uy*j + uz*k
+can be represented as a quaternion:
+q = cos(theta/2) + sin(theta/2) * [ux*i + uy*j + uz*k]
+i.e.:
+qw = cos(theta/2)
+qx = sin(theta/2) * ux
+qy = sin(theta/2) * uy
+qz = sin(theta/2) * uz
+
+For a quaternion to represent a rotation or orientation, it must be unit-length.
+
+References
+----------
+.. https://github.com/matthew-brett/transforms3d/blob/master/transforms3d/quaternions.py
+.. http://mathworld.wolfram.com/Quaternion.html
+"""
+
+from compas.geometry.transformations import quaternion_from_matrix
+from compas.geometry.transformations import matrix_from_quaternion
+from compas.geometry.transformations import euler_angles_from_matrix
+from compas.geometry.transformations import matrix_from_euler_angles
+from compas.geometry.transformations import axis_and_angle_from_matrix
+from compas.geometry.transformations import matrix_from_axis_and_angle
+
+from compas.geometry.basic import allclose
+
+import math
+
+
+__all__ = [
+    'quaternion_norm',
+    'quaternion_unitize',
+    'quaternion_is_unit',
+    'quaternion_multiply',
+    'quaternion_canonize',
+    'quaternion_conjugate',
+    'quaternion_from_euler_angles',
+    'euler_angles_from_quaternion',
+    'quaternion_from_axis_angle',
+    'axis_angle_from_quaternion']
+
+ATOL = 1e-8  # absolute tolerance
```

Comment:
should we add this to `compas`? => `compas.ATOL` like with the precision variable `compas.PRECISION`. it seems to make sense to control this "COMPAS-wide"...

Reply:
@gonzalocasas 
@romanarust

Reply:
on the one hand it would be handy, on the other - I doubt you one number will fit all cases. You might want a different precision (assuming SI units only anyway) for e.g. meters than radians than Newtons.. ?

Reply:
I would go for `compas` wide control, but agree with @funkchaser: different tolerance values for different measures (`compas.ATOL_ANGLE`, ...)